pkg/virt-controller/services/template.go

```diff
@@ -47,12 +49,29 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VM) (*kubev1.Pod, error) {
 	domain := precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
 	uid := precond.MustNotBeEmpty(string(vm.GetObjectMeta().GetUID()))
 
+	// Compute resource requirements taking the VM overhead into account
+	memoryRequired, e := resource.ParseQuantity(
+		fmt.Sprintf("%dKi", vm.Spec.Domain.ComputeTotalMemoryRequirements()))
+	if e != nil {
+		return nil, e
+	}
+
+	resources := kubev1.ResourceRequirements{
+		Requests: kubev1.ResourceList{
+			kubev1.ResourceMemory: memoryRequired,
+		},
+		Limits: kubev1.ResourceList{
```

Comment:
I think it must be set to `kubev1.ResourceMemory: vm.Spec.Domain.Memory.Maximum` if we want to allow hotplug and overcommitment,

Reply:
That is possible but not at all certain. It is possible we will be creating a new Pod for the VM when doing hotplug. At least I remember hearing that discussion some time ago.

Reply:
It seems doubtful that a new pod will be involved with hotplug.

It makes sense to be for the container mem request to be (Required + overhead) and the container mem limit to be (Maximum + overhead)

Both Required and Maximum are related to the guest vm only (right?), so they both need the overhead added to them when translating this to container memory limits.

Reply:
There are many things to the details - i.e. calculating the overhead.

For now, let's start simple with basic request fields, which specify the minimum requirements.

Form there we can see how to optimize it.

Reply:
Minimum requirements are not enough when scheduling VMs. Because you won't know if the VM will have enough memory once something else starts eating host memory. The memory has to be guaranteed or the VMs will start crashing (the guest can't easily react on host memory pressure, it is too isolated for that).